plugins/titleDiff/titleDiff.js

```diff
@@ -0,0 +1,73 @@
+var TAG_NAME = "_titleDiff" // Name of tag to set when scene title differs from filename, will create if not found
+
+
+function main() {
+    switch (input.Args.mode) {
+        case "diff":
+            diff()
+    }
+}
+
+function diff() {
+    log.Info("diff: Begin diff title/filename")
+    var sceneIds = findDiffingScenes()
+    var tagId = getOrCreateTag()
+    var n = setTag(sceneIds, [tagId])
+
+    log.Info("diff: Number of scenes found with differing title/filename: " + n)
+}
+
+function findDiffingScenes() {
+    var query = "{findScenes(filter:{per_page:-1}){scenes{id title path}}}"
+    var result = gql.Do(query)
+    if (result.findScenes == null) {
+        log.Info("diff: No scenes found")
+        return
+    }
+    var scenes = result.findScenes.scenes
+    var diffs = []
+    for (var i = 0; i < scenes.length; i++) {
+        var filename = scenes[i].path.split(/.*[\/|\\]/)[1];
+        var filenameNoExt = filename.replace(/\.[^/.]+$/, "")
```

Comment:
Can you add an extra check for future proofing here?
With the <https://github.com/stashapp/stash/pull/3006> PR there can be scenes with no files. For those the path is empty and thus you get `Plugin returned error: TypeError: Cannot access member 'replace' of undefined ` and the plugin exits. For those cases i think just skipping the scene is ok